emr_run_spark.py

```diff
@@ -192,19 +193,46 @@ def create_cluster_and_run_job_flow(create_cluster_master_type=None,
                           send_success_email_to=send_success_email_to)
     client = _get_client(aws_region)
     debug_steps = _create_debug_steps(create_cluster_setup_debug)
-    response = client.run_job_flow(
-        Name=job_flow_name,
-        LogUri=s3_logs_uri,
-        ReleaseLabel='emr-4.6.0',
-        Instances={
+    if bid_price:
+        instances = {
+                'InstanceGroups': [
+                    {
+                        'Name': 'EmrMaster',
+                        'Market': 'SPOT',
```

Comment:
Should the master really be spot? Can you check what's the best practice for that?